pkg/local_object_storage/metabase/iterators.go

```diff
@@ -0,0 +1,152 @@
+package meta
+
+import (
+	"strconv"
+
+	"github.com/nspcc-dev/neofs-api-go/pkg/container"
+	"github.com/nspcc-dev/neofs-api-go/pkg/object"
+	objectV2 "github.com/nspcc-dev/neofs-api-go/v2/object"
+	"github.com/pkg/errors"
+	"go.etcd.io/bbolt"
+)
+
+// ExpiredObject is a descriptor of expired object from DB.
+type ExpiredObject struct {
+	typ object.Type
+
+	addr *object.Address
+}
+
+// Type returns type of the expired object.
+func (e *ExpiredObject) Type() object.Type {
+	return e.typ
+}
+
+// Address returns address of the expired object.
+func (e *ExpiredObject) Address() *object.Address {
+	return e.addr
+}
+
+// ExpiredObjectHandler is an ExpiredObject handling function.
+type ExpiredObjectHandler func(*ExpiredObject) error
+
+// ErrInterruptIterator is returned by iteration handlers
+// as a "break" keyword.
+var ErrInterruptIterator = errors.New("iterator is interrupted")
+
+// IterateExpired iterates over all objects in DB which are out of date
+// relative to epoch.
+//
+// If h returns ErrInterruptIterator, nil returns immediately.
+// Returns other errors of h directly.
+func (db *DB) IterateExpired(epoch uint64, h ExpiredObjectHandler) error {
+	return db.boltDB.View(func(tx *bbolt.Tx) error {
+		return db.iterateExpired(tx, epoch, h)
+	})
+}
+
+func (db *DB) iterateExpired(tx *bbolt.Tx, epoch uint64, h ExpiredObjectHandler) error {
+	err := tx.ForEach(func(name []byte, b *bbolt.Bucket) error {
+		cidBytes := cidFromAttributeBucket(name, objectV2.SysAttributeExpEpoch)
+		if cidBytes == nil {
+			return nil
+		}
+
+		return b.ForEach(func(expKey, _ []byte) error {
+			bktExpired := b.Bucket(expKey)
+			if bktExpired == nil {
+				return nil
+			}
+
+			return bktExpired.ForEach(func(idKey, _ []byte) error {
+				expiresAt, err := strconv.ParseUint(string(expKey), 10, 64)
```

Comment:
We don't need to calculate `expiresAt` and make an epoch check on each record in `bktEpxired` bucket. We can move it higher:

```go
			bktExpired := b.Bucket(expKey)
			if bktExpired == nil {
				return nil
			}
			expiresAt, err := strconv.ParseUint(string(expKey), 10, 64)
			...
```

Reply:
Done